test/test_archive_cleanup.py

```diff
@@ -0,0 +1,113 @@
+
+import datetime
+import json
+import sys
+from unittest import mock
+
+import pytest
+
+from pghoard import archive_cleanup
+
+
+def test_missing_config_arg():
+    args = ["archive_cleanup", "--dry-run"]
+    tool = archive_cleanup.ArchiveCleanup()
+    with mock.patch.object(sys, "argv", args):
+        assert tool.run() == 1
+
+
+def test_main_invalid_config():
+    args = ["archive_cleanup", "--config", "/etc/os-release", "--dry-run"]
+    with mock.patch.object(sys, "argv", args):
+        assert archive_cleanup.main() == 1
+
+
+def test_invalid_config():
+    args = ["archive_cleanup", "--config", "/etc/os-release", "--dry-run"]
+    tool = archive_cleanup.ArchiveCleanup()
+    with mock.patch.object(sys, "argv", args):
+        with pytest.raises(archive_cleanup.InvalidConfigurationError):
+            tool.run()
+
+
+def test_set_config():
+    tool = archive_cleanup.ArchiveCleanup()
+    tool.set_config("pghoard.json", "default")
+    assert tool.site == "default"
+    assert "basebackup_count" in tool.backup_site
+    assert "backup_location" in tool.config
+    assert tool.storage is not None
+
+
+def test_archive_cleanup(tmp_path):
+    tool = archive_cleanup.ArchiveCleanup()
+    with open("pghoard-local-minimal.json", "r") as f:
+        config = json.load(f)
+    config["backup_sites"]["example-site"]["object_storage"]["directory"] = (tmp_path / "backups").as_posix()
+    config["json_state_file_path"] = (tmp_path / "pghoard_state.json").as_posix()
+    config_path = tmp_path / "config.json"
+    with open(config_path, "w") as f:
+        json.dump(config, f, indent=4)
+
+    bb_metadata = {
+        "backup-decision-time": "2022-03-23T14:57:55.883514+00:00",
+        "backup-reason": "scheduled",
+        "start-time": "2022-03-23T15:57:55+01:00",
+        "start-wal-segment": "000000010000000000000002",
+        "active-backup-mode": "basic",
+        "pg-version": "130006",
+        "compression-algorithm": "snappy",
+        "compression-level": "0",
+        "original-file-size": "25933312",
+        "host": "toolbox"
+    }
+    bb_path = tmp_path / "backups" / "example-site" / "basebackup"
+    bb_path.mkdir(parents=True)
+    (bb_path / "2022-03-23_14-57_0").touch()
+    with open((bb_path / "2022-03-23_14-57_0.metadata").as_posix(), "w") as f:
+        json.dump(bb_metadata, f, indent=4)
+
+    xlog_metadata = {
+        "pg-version": "130006",
+        "compression-algorithm": "snappy",
+        "compression-level": "0",
+        "original-file-size": "16777216",
+        "host": "toolbox",
+        "hash": "d6398f89d3dbf9ce8f68bbe5c07cc0208415a8ff",
+        "hash-algorithm": "sha1"
+    }
+
+    xlog_path = tmp_path / "backups" / "example-site" / "xlog"
+    xlog_path.mkdir()
+    (xlog_path / "000000010000000000000001").touch()
+    with open((xlog_path / "000000010000000000000001.metadata").as_posix(), "w") as f:
+        json.dump(xlog_metadata, f, indent=4)
```

Comment:
nit. why the "manual" open when using a `Path` object ?
You could just do: 
```python
with (xlog_path / "000000010000000000000001.metadata").open("w") as f:
   json.dump(xlog_metadata, f, indent=4)
```
Or even:
```python
(xlog_path / "000000010000000000000001.metadata").write_text(json.dumps(xlog_metadata, indent=4))
```
Or do we need something very specific that requires using `as_posix` ?

Reply:
Ah right, didn't think about it (and wasn't aware of `write_text()`).